azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/module_client.py

```diff
@@ -4,7 +4,19 @@
 # --------------------------------------------------------------------------------------------
 
 from .internal_client import InternalClient
+from .message import Message
 
 
 class ModuleClient(InternalClient):
-    pass
+    def send_to_output(self, message, output_name):
+        """
+        Sends an event/message to the given module output.
+        This is an outgoing events and  means to be "output events"
+        :param output_name: Name of the output to send the event to.
+        :param message: message to send to the given output
+        """
+        if not isinstance(message, Message):
+            message = Message(message)
```

Comment:
>message = Message(message) [](start = 12, length = 26)

If send_to_output does this at the client level, then internal_client.send_event should do this at the client level for symmetry sake.  (I think we have code at a lower level that does the same wrapping when internal_client.send_event is called)

Reply:
yes there is a code to wrap it in message before encoding the properties. The reason the wrapping was needed in module client is because we are already accessing the output name here. We should have symmetry and put it in internal client as well. If that's the case then we will no longer need it in transport. 

---
In reply to: [246253488](https://github.com/Azure/azure-iot-sdk-python-preview/pull/35#discussion_r246253488) [](ancestors = 246253488)